featuretools/synthesis/encode_features.py

```diff
@@ -118,9 +123,12 @@ def encode_features(feature_matrix, features, top_n=10, include_unknown=True,
         val_counts = val_counts.sort_values([f.get_name(), index_name],
                                             ascending=False)
         val_counts.set_index(index_name, inplace=True)
-        select_n = top_n
+        f_top_n = top_n
+        if isinstance(top_n, dict):
+            f_top_n = top_n.get(f.get_name(), DEFAULT_TOP_N)
+        select_n = f_top_n
```

Comment:
if `top_n` is not a dict the logic looks like
```python
f_top_n = top_n
select_n = f_top_n
```

I think we can simplify by getting removing the `f_top_n` variable